From 41a5d97e318d95f8feab41d9911001280597132c Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Sun, 21 Oct 2018 13:30:31 -0700
Subject: [PATCH 3/4] Format match arms in vertical style

By default rustfmt formats match arms in a way to prefer horizontal
layout over vertical one. E.g.,
  > match key {
  >   Key::Esc | Key::Return => {
  >     ...
  >   },

Most of the time, however, a vertical layout is preferred as it allows
for easier addition/removal of lines and generally (although
subjectively) provides for a quicker overview of the alternatives. For
example:
  > match key {
  >   Key::Esc |
  >   Key::Return => {
  >     self.in_out = InOut::Clear;

This change adjusts rustfmt to format match arms in such a vertical
style.
---
 src/tools/rustfmt/src/expr.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/rustfmt/src/expr.rs b/src/tools/rustfmt/src/expr.rs
index 74639c..9a497c 100644
--- a/src/tools/rustfmt/src/expr.rs
+++ b/src/tools/rustfmt/src/expr.rs
@@ -1191,7 +1191,7 @@ pub fn rewrite_multiple_patterns(
     } else {
         definitive_tactic(
             &items,
-            ListTactic::HorizontalVertical,
+            ListTactic::Vertical,
             Separator::VerticalBar,
             shape.width,
         )
-- 
2.18.1

